src/funding/funding.js

```diff
@@ -29,7 +33,7 @@ export function isFundingIneligible(source : FundingSource, { locale, funding, l
         return FUNDING_ELIGIBILITY_REASON.DISALLOWED_COUNTRY;
     }
 
-    if (getFundingConfig(source, 'requireCommitAsTrue') && !commit) {
+    if ((getFundingConfig(source, 'requireCommitAsTrue') && !commit) || config.apmTestDomains.indexOf(apmDomain) !== -1) {
```

Comment:
I'm not sure this boolean logic is correct -- it should be an `&&`. Can we do this:

```javascript
if ((getFundingConfig(source, 'requireCommitAsTrue') && !commit) {
    if (config.apmTestDomains.indexOf(apmDomain) !== -1)) {
        return FUNDING_ELIGIBILITY_REASON.COMMIT_NOT_SET;
    }
}
```

Can we also hoist the domain check outside of this function? Otherwise we're looping that array for each individual funding source.

Reply:
So the current logic is if the merchant is whitelisted, the APMs button should be rendered regardless commit passed as true or false

Reply:
In which case, this should be `&& !isWhitelistedMerchant`.